homeassistant/components/telegram_bot/webhooks.py

```diff
@@ -98,10 +122,12 @@ async def post(self, request):
             return self.json_message("Access denied", HTTPStatus.UNAUTHORIZED)
 
         try:
-            data = await request.json()
+            update_data = await request.json()
         except ValueError:
             return self.json_message("Invalid JSON", HTTPStatus.BAD_REQUEST)
 
-        if not self.process_message(data):
-            return self.json_message("Invalid message", HTTPStatus.BAD_REQUEST)
+        update = Update.de_json(update_data, self.bot)
+        _LOGGER.debug("Received Update on %s: %s", self.url, update)
+        self.dispatcher.process_update(update)
```

Comment:
Good questions, I'll look in to it when I get the chance 👍

Reply:
Looks like we don't rely on any threading in the telegram library for the webhook setup. I guess since telegram doesn't care about the outcome we can process the update with `async_add_executor_job`.